06_puppy/jimbotech/mapStore_test.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type storesSuite struct {
+	suite.Suite
+	store  Storer
+	mapper mapCheck
+}
+
+func TestStorer(t *testing.T) {
+	m := NewPuppyStorer()
+	ms := &SyncMapStore{}
+	suite.Run(t, &storesSuite{store: m, mapper: m})
+	suite.Run(t, &storesSuite{store: ms, mapper: ms})
+}
+
+// TestMapStoreWithoutContructor may not be required as it may not be
+// possible to create an instance of that type without using the constructor
+func TestMapStoreWithoutContructor(t *testing.T) {
```

Comment:
If you split up the files as I suggested earlier, this function would belong in `mapstore_test.go` - assuming you even keep the `ErrNotConstructed` error.

Reply:
Split doen